src/modules/app/app.ts

```diff
@@ -29,36 +34,97 @@ export class App {
                     value,
                     isAdd
                 ),
-            (param: string, min: string, max: string) => this.searchParams.updateSearchParamBySlider(param, min, max),
-            (param: string, value: string) => this.searchParams.updateSearchParamBySearch(param, value)
+            (param: string, min: string, max: string) =>
+                this.searchParams.updateSearchParamBySlider(param, min, max),
+            (param: string, value: string) =>
+                this.searchParams.updateSearchParamBySearch(param, value)
         );
-        const products: Element = document.querySelector('.products')!; // ???
+
         this.productList = new ProductList(
-            products,
+            PRODUCTS_LIST_CONTAINER,
             (id: number, isAdded: boolean) =>
-                this.onProductItemSelected(id, isAdded)
+                this.onProductItemSelected(id, isAdded),
+            (id: number) => this.onProductClick(id)
         );
-        const header: Element = document.querySelector('.header__container')!; //???
 
-        this.cart = new Cart(header);
+        this.cart = new Cart(HEADER_CONTAINER);
     }
 
     public start(): void {
         this.cart.drawCart();
-        this.cart.setCurrentValues('0', '0');
+        this.cart.setCurrentValues(this.getSum(), this.getCount());
         this.cart.resetBtn(); // will be remove at the end
+
+        this.drawPageByUrl(window.location.href, window.location.search);
+    }
+
+    private drawPageByUrl(url: string, search: string): void {
+        if (url.includes('id')) {
+            const id: number = Number(url.split('=').pop());
+            this.drawProductPage(id);
+            return;
+        }
+
         this.globalFiltres.createFilters(productsData);
+        const searchParams = Object.fromEntries(new URLSearchParams(search));
 
-        const startFilters: Map<FilterType, Set<string>> = this.globalFiltres.getCurrentFilters();
-        const startSliders: Map<SliderType, SliderValue> = this.globalFiltres.getCurrentSliders();
-        this.updateResult(startFilters, startSliders, '');
+        if (search) {
```

Comment:
When parameter is not `boolean` I would suggest this type of `if`. Is it comparison against `null`, `undefined`, `empty` or something else? Make it explicit.

Reply:
Fixed.